### ISSUE TITLE
Rust Syntax highlighting fails on unicode-declaring string literals.

### ISSUE THREAD

**Issue by llogiq** [geany/geany#1089]
Example:
```
let x = "\u{FFFE}"; // <-- Everything from here on will be colored as string literal
let y = ..
```

The problem appears to be that the syntax highlighter cannot deal with the squiggly braces after the `\u`, if I remove them, the syntax highlighter colors the rest of the code correctly. However, without the braces, the string literal is no longer valid Rust.

**Comment by elextr** [comment]
The lexer is part of the Scintilla project which provides the editing component we (and others) use.  You should report it to them [here](https://sourceforge.net/p/scintilla/bugs/)

**Comment by llogiq** [comment]
I believe it was already fixed in Scintilla [here](https://sourceforge.net/p/scintilla/bugs/1809/) – so why does the fix apparently not apply to Geany?

**Comment by b4n** [comment]
@llogiq the fix is not yet available in a released version of Geany.  Next release due soon will included it (it upgrades from Scintilla 3.6.3 to 3.6.6, and the fix was introduced in 3.6.4).

**Comment by llogiq** [comment]
Thank you.
